### DISCUSSION TITLE
Body defined with 1 singular value cause fail

### DISCUSSION THREAD

**Discussion by drnextgis** [fastapi/fastapi#6847] [Questions] [answered]
Is there a way to define JSON body with only single key using `Body`? It works fine when body has more then 1 key, for example:

```python
import uvicorn
from fastapi import Body, FastAPI

app = FastAPI()


@app.post("/good")
async def good(name: str = Body(...), surname: str = Body(...)):
    return {"name": name, "surname": surname}

uvicorn.run(app, host='127.0.0.1', port=8000)
```

```bash
$ curl -X POST "http://127.0.0.1:8000/good" -d '{"name": "John", "surname": "Doe"}'
{"name":"John","surname":"Doe"}
```

But I couldn't make it work for body with only one key:
```python
@app.post("/bad")
async def bad(name: str = Body(...)):
    return {"name": name}
```

```bash
$ curl -X POST "http://127.0.0.1:8000/bad" -d '{"name": "John"}'                  
{"detail":[{"loc":["body"],"msg":"str type expected","type":"type_error.str"}]}
```
Am I missing something?

**Comment by Kludex** [comment]
You are.
 
Try this:
```bash
$ curl -X POST "http://127.0.0.1:8000/bad" -d '{"John"}'                  
```

**Comment by drnextgis** [comment]
@Kludex thanks but it doesn't work either. I have found an answer [here](https://stackoverflow.com/a/61986099/813758):

```python
@app.post("/bad")
async def bad(name: str = Body(..., embed=True)):
    return {"name": name}
```

```bash
$ curl -X POST "http://127.0.0.1:8000/bad" -d '{"name": "John"}'
{"name":"John"}
```

**Comment by Kludex** [comment, marked as answer]
Sorry, I meant (without the brackets) : :sweat_smile:
```bash
$ curl -X POST "http://127.0.0.1:8000/bad" -d '"John"'                  
```

But you can also use the embed, I just wanted to show you that it's not wrong, you just expect a different input format.

**Comment by tiangolo** [comment]
Thanks for the help here @Kludex ! 👏  🙇 

Thanks for reporting back and closing the issue @drnextgis  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
